### PR TITLE
Moved Method Does Not Take (More) Parameters error to case class

### PR DESCRIPTION
**Pull request by ennru**
@felixmulder Same solution regarding `typer.ErrorReporting.Errors` here.

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala [hunk 1]
```diff
@@ -1223,4 +1223,22 @@ object messages {
...
+      case _ => ""
+    }
+
+    val msg = hl"${err.refStr(methPartType)} does not take$more parameters"
+
+    val kind = "Reference"
+
+    val explanation =
+      hl"""|You have specified more parameter lists as defined in the method definition(s).
+           |In case ${err.refStr(methPartType)} is defined without parenthesis, you may
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala]: You should be able to only display this when `methPartType` is actually defined without parenthesis. I think `methPartType.isInstanceOf[ExprType]` should check for that.

Comment (ennru) [inline compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala]: Great, was looking around for an easy enough check to do just that.

Comment (ennru) [inline compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala]: @smarter I find a `TermRefWithSignature` in my samples, do I need to test for just that?

Comment (smarter) [inline compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala]: Try `methPartType.widenSingleton.isInstanceOf[ExprType]`

Comment (smarter) [inline compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala]: (A `TermRef` might not always be a reference to a method with no arguments, so we widen it to its underlying type)

**Comment by felixmulder** [comment]
Also needs rebase, then LGTM

**Comment by felixmulder** [comment]
@dotty-bot: could you recheck please?

**Pull request merged**